packages/create-amplify/src/logger.ts

```diff
@@ -1,63 +1,138 @@
 import yargs from 'yargs';
+import * as os from 'os';
 
 /**
  * A logger that logs messages to the console.
  */
 export class Logger {
   /**
-   * Creates a new Logger instance.
-   * @param console The console to log to.
+   * Properties for ellipsis animation
+   */
+  private timer: ReturnType<typeof setTimeout>;
+  private refreshRate: number;
+  private animateMessage: string;
+
+  /**
+   * Creates a new Logger instance. Injecting stdout for testing.
+   * @param stdout The stream to write logs to.
    * @param minimumLogLevel The minimum log level to log.
    */
   constructor(
-    private readonly console: Console = global.console,
+    private readonly stdout = process.stdout,
     private readonly minimumLogLevel: LogLevel = LogLevel.INFO
-  ) {}
+  ) {
+    this.refreshRate = 500; // every 0.5 seconds
+  }
 
   /**
    * Logs a message to the console.
    */
-  async log(message: string, level: LogLevel = LogLevel.INFO) {
+  log(message: string, level: LogLevel = LogLevel.INFO) {
     const toLogMessage = level <= this.minimumLogLevel;
 
     if (!toLogMessage) {
       return;
     }
 
-    const logMessage = `[${
-      LogLevel[level]
-    }] ${new Date().toISOString()}: ${message}`;
-    this.console.log(logMessage);
+    const logMessage =
+      this.minimumLogLevel === LogLevel.DEBUG
+        ? `[${LogLevel[level]}] ${new Date().toISOString()}: ${message}`
+        : message;
+    this.stdout.write(logMessage + os.EOL);
+  }
+
+  /**
+   * Start animating ellipsis at the end of a log message.
+   */
+  startAnimatingEllipsis(message: string) {
```

Comment:
this doesn't account for someone calling this twice without stopping the first one. If a timer is already set, we should probably throw and say to stop the first one

Reply:
Done in [26a70ce](https://github.com/aws-amplify/amplify-backend/pull/595/commits/26a70ceb7f1d7ebd88108f9e7c3736df98e97448)